src/component.js

```diff
@@ -1,75 +1,69 @@
-import { copyObject } from '../utils/tools'
-import { h } from 'vue'
+import { copyObject } from './utils';
+import { h } from 'vue';
 
 function destroyChart () {
-  if (this.chart) {
-    this.chart.destroy()
-  }
-}
-
-const generateVueComponent = function (Highcharts, VueVersion) {
-  const VUE_MAJOR = VueVersion.split('.')[0]
-  const VERSION_DEPENDENT_PROPS = VUE_MAJOR < 3
-    ? {
-    // Fallback options for Vue v2 to keep backward compatibility.
-      render: (createElement) => createElement('div', {
-        ref: 'chart'
-      }),
-      beforeDestroy: destroyChart
-    // The new Vue's 3 syntax.
-    } : {
-      render () { return h('div', { ref: 'chart' }) },
-      beforeUnmount: destroyChart
+    if (this && this.chart && this.chart.destroy) {
```

Comment:
Is there any reason of why we couldn't use optional chaining here?
Also, since this function is no longer re-used, we could consider moving its body directly into the component. What do you think?